gui/functionality/idact_notebook.py

```diff
@@ -160,71 +152,70 @@ def deploy_notebook(self):
             sleep_until_allocation_ends(nodes=nodes)
         return
 
-    def open_new_native_argument(self):
-        self.add_argument_window.show()
-    
-    def open_remove_native_argument(self):
-        self.remove_argument_window.show()
-    
-    def open_show_native_argument(self):
-        native_args_list = self.native_args_saver.get_native_args_list()
-        counter = len(native_args_list)
-        self.show_native_arguments_window.ui.table_widget.setRowCount(counter)
-        self.show_native_arguments_window.ui.table_widget.setColumnCount(2)
-
-        for i in range(counter):
-            self.show_native_arguments_window.ui.table_widget.setItem(i, 0, QTableWidgetItem(native_args_list[i][0]))
-            self.show_native_arguments_window.ui.table_widget.setItem(i, 1, QTableWidgetItem(native_args_list[i][1]))
-        self.show_native_arguments_window.show()
-
-    def add_new_native_argument(self):
-        argument_name = self.add_argument_window.ui.argument_name_edit.text()
-        self.parameters['add_native_arguments']['argument_name'] = argument_name
-        value = self.add_argument_window.ui.value_name_edit.text()
-        self.parameters['add_native_arguments']['value'] = value
-        self.saver.save(self.parameters)
+    def prepare_native_args(self, args):
+        prepared_args = args.copy()
+        keys = list(prepared_args)
 
-        self.native_args_saver.add_to_list(('--'+argument_name, value))
+        for key in keys:
+            if not key.startswith('--'):
+                prepared_args['--'+key] = prepared_args.pop(key)
+        return prepared_args
 
-    def remove_native_argument(self):
-        argument_name = self.remove_argument_window.ui.argument_name_edit.text()
-        self.parameters['remove_native_arguments']['argument_name'] = argument_name
-        self.saver.save(self.parameters)
+    def open_edit_native_argument(self):
+        native_args = self.native_args_saver.get_native_args()
```

Comment:
Why are you using here the `self.native_args_saver.get_native_args()` method but in the `prepare_native_args` you use this outside the function?

Reply:
I'm not sure what's the point. Could you explain it once again?

Reply:
For me it's just inconsistent. In line 165 you use the `self.native_args_saver.get_native_args()` inside a function, and in line 128 you use it outside the function. Just that.

Reply:
Ok, i fixed it.